mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -151,17 +377,19 @@ static LogicalResult generateLoopNestUsingForOp(
   assert(loopRanges.size() == tileSizes.size() &&
          "expected as many tile sizes as loop ranges");
   OpBuilder::InsertionGuard guard(rewriter);
-  SmallVector<Value> ivs;
 
-  for (auto [loopRange, tileSize] : llvm::zip_equal(loopRanges, tileSizes)) {
-    // No loops if tile size is zero. Set offset and size to the loop
-    // offset and size.
-    if (isConstantIntValue(tileSize, 0))
-      continue;
+  SmallVector<OpFoldResult> lbs, ubs, steps;
+  std::tie(lbs, ubs, steps) =
+      getLoopBounds(rewriter, loc, loopRanges, tileSizes);
```

Comment:
this goes to OpFoldResult to immediately go to Value, can these be fused ?

Reply:
If `scf.for` took `OpFoldResult` maybe. `scf.forall` does. So I think it make sense for `getLoopBounds` to return `SmallVector<OpFoldResult>`s and it gets materialized to a value in the `scf.for` creation method cause this is needed because of `scf.for`.